src/ScraperBot/Plugin/README.md

```diff
@@ -0,0 +1,154 @@
+project root called `active_plugin_store.yaml`. Longer term, alternative storage mechanisms could be defined since the 
+store itself is swappable.
+
+The `active_plugin_store.yaml` file format consists of the plugin type as a key, followed by an array of plugin IDs 
+that are active.
+
+For example:
+
+```
+storage:
+  - sqlite3
+  - mysql
+``` 
+
+## Creating a plugin
+
+To create a new plugin, you must know two pieces of information.
+
+1. The 'type' of the plugin.
+2. The plugin's interface that you must implement. 
+
+For example, a storage plugin has the type `storage` and implements the `ScraperBot\Storage\StorageInterface` interface.
+
+The plugin 'type' can be found in the `*.plugin.type.yml` file for that plugin type, and uses the key `type`.
+
+The interface you must implement can be found in the same file with the `interface` key.
+
+Another place to conveniently find both of these pieces of information is at the `/plugins` URL, where you can see the 
+list of registered plugin types, as well as the metadata such as the interface that a plugin of that type must 
+implement.  At this URL you can also see a list of currently active plugins.
+
+**Plugins must live sub-folders of the `custom` folder. This name becomes part of your class namespace, so it is
+important to choose this name with that in mind.**
+
+Once you have this information, you need to do two things.
+
+1. Create a class which implements the interface defined by the type you are implementing.  
+    - The file should live under a `src' folder that sits alongside the *.plugins.yml file you'll define in the next step.
+    - The namespace prefix should consist of `\ScraperBot` and the parent directory of the file created in the next step. 
+2. Define a <type>/plugins.yml file and reference your new class there using the fully qualified name as the `class` 
+key.
+
+Plugins in the *.plugins.yml file are keyed by ID, and you can define multiple instances.
+
+For example, the core storage plugin is defined in `storage.plugins.yml` and looks like
+
+```
+sqlite3:
+  class: ScraperBot\Storage\Plugin\SqlLite3Storage
+  description: SqlLite3 storage plugin
+```
+
+When adding custom plugins under the 'custom' folder, the autoloader is automatically updated based on the location of
+the discovered `*.plugins.yaml` file.
+
+As mentioned earlier, the parent folder of your `*.plugins.yaml` file will be used as
+part of the namespace for loading of your custom classes. There should be a 'src' folder alongside your yaml file that
+contains your classes.
+
+For example, your structure might be
+
+`custom/MySqlStorage/storage.plugins.yaml`
+
+The contents might be
+
+```
+mysql:
+  class: ScraperBot\MySqlStorage\Plugin\Storage\MySqlStorage
+  description: MySQL storage plugin
+```
+
+Notice the namespace of the class. Such namespaces always use `ScraperBot\` as a prefix, followed by the folder name.
```

Comment:
not sure we should change this to GlitcherBot instead of ScraperBot ? What do you think?

Reply:
Yeah, we already have a lot of legacy stuff in terms of the name change!  For now, I'd keep it consistent and then maybe undertake a task to update namespaces globally?